Wait a configurable amount of time for benchmark channels to connect

apolcyn opened the pull request:
Needed for an internal benchmark in which the server takes a while to become ready because of the time it takes for service discovery mechanisms to come up. This fixes the situation on that end.

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,356      Total (=)      1,976,356

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,770,407      Total (=)     10,770,407

 No significant differences in binary sizes


```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,356      Total (=)      1,976,356

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,770,413      Total (>)     10,770,408

 No significant differences in binary sizes


```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

ncteisen commented:
How long does it usually take to become connected? If it is under a minute I think we should just increase the timeout, rather than add yet another configuration environment variable

apolcyn commented:
At least from what I've seen so far, it's commonly around 2 minutes. What do you think?

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

grpc-testing commented:
```
[trickle] No significant performance differences
```

apolcyn commented:
Good point to document it. I've added a comment, but I'm having a thought about adding it to `doc/environment-variables.md`: it looks like that md file <i>only</i> currently contains environment variables that affect the C-Core production library - test code, C++, etc..... so I've added documentation to `tools/run_tests/performance/README.md` instead. 

That look ok?

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,356      Total (=)      1,976,356

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,770,403      Total (<)     10,770,409

 No significant differences in binary sizes


```

grpc-testing commented:
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,356      Total (=)      1,976,356

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,770,409      Total (<)     10,770,412

 No significant differences in binary sizes


```

grpc-testing commented:
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,356      Total (=)      1,976,356

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,770,412      Total (>)     10,770,410

 No significant differences in binary sizes


```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

jtattermusch commented on tools/run_tests/performance/README.md:
```diff
@@ -104,3 +104,29 @@ Example memory profile of grpc-go server, with `go tools pprof`:
+  Type: comma separated list of host:port
+
+  Set this to a comma separated list of QPS worker processes/machines.
+  Note that the driver will start the "benchmark server" on the first
```
I think this is not accurate.  from the list, the first N items will be consumed to represent servers based on the json configuration of how many servers are requested for given scenario. The next K will be taken as clients).
https://github.com/grpc/grpc/blob/158fb288fc9807ddf9d6adca6f3d4803e855ae34/tools/run_tests/performance/scenario_config.py#L137

special cases:
- if you specify num_server + num_clients  less than the number of items in QPS_WORKERS, the rest will get unused.
- i think num_clients=0 used to mean something like "consume all the remaining workers from the list", but I'm not sure if that's the case or not.

It's actually pretty convoluted :-)

apolcyn replied:
good catch :) fixed

grpc-testing commented:
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

grpc-testing commented:
```
[trickle] No significant performance differences
```

grpc-testing commented:
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,976,356      Total (=)      1,976,356

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,770,404      Total (<)     10,770,409

 No significant differences in binary sizes


```

grpc-testing commented:
```
[microbenchmarks] No significant performance differences
```

apolcyn commented:
tests failures: TSAN C++ https://github.com/grpc/grpc/issues/16124
objc dbg and opt: https://github.com/grpc/grpc/issues/16156

adelez commented on tools/run_tests/performance/README.md:
```diff
@@ -104,3 +104,31 @@ Example memory profile of grpc-go server, with `go tools pprof`:
+  Type: comma separated list of host:port
+
+  Set this to a comma separated list of QPS worker processes/machines.
+  Each scenario in a scenario config has specifies a certain number
```
nit: Remove 'has'
